server/bingoServer.js

```diff
@@ -0,0 +1,131 @@
+const express = require("express");
+const app = express();
+const server = require('http').createServer(app);
+const { Server, Socket } = require("socket.io");
+const { serverEvents, clientEvents } = require('./events');
+const { generateBingoCard } = require("../utils/bingo/cardGenerator");
+const { generateGameSequence, checkForWin } = require("../utils/bingo/game");
+const { setInterval } = require("timers/promises");
+const BINGO_NUMBERS_DELAY = 4000
+
+class BingoServer {
+    constructor(io) {
+        this.gameState = {
+            mode: 'NORMAL',
+            players: new Map(), // players[socket.id] = { id, name, table }
+            inProgress: false,
+            numbers: generateGameSequence(),
+        };
+        this.io = io;
+        this.intervalId = null;
+        this.listenToClientEvents();
+    }
+
+    listenToClientEvents() {
+        this.io.on("connection", (socket) => {
+            socket.on("disconnect", () => {
```

Comment:
Esto bastará para saber si un usuario sigue en linea?

Reply:
Con esta definición, estar el línea significa que el cliente se mantiene conectado al servidor (envia PING, espera PONG) https://socket.io/docs/v4/how-it-works/